pep-0723.rst

```diff
@@ -0,0 +1,185 @@
+      "requests<3",
+      "rich",
+    ]
+    """
+
+    import requests
+    from rich.pretty import pprint
+
+    resp = requests.get("https://peps.python.org/api/peps.json")
+    data = resp.json()
+    pprint([(k, v["title"]) for k, v in data.items()][:10])
+
+
+Reference Implementation
+========================
+
+This regular expression may be used to parse the metadata::
+
+    (?ms)^__pyproject__ *= *"""$(.+?)^"""$
+
+For languages that do not support easily accessing the match group, one may
+parse the entire match as TOML and then access the ``__pyproject__`` key.
+
+The following is an example of how to read the metadata on Python 3.11 or
+higher.
+
+.. code:: python
+
+    import re, tomllib
+
+    def read(script: str) -> dict | None:
+        match = re.search(r'(?ms)^__pyproject__ *= *"""$(.+?)^"""$', script)
+        return tomllib.loads(match.group(1)) if match else None
+
+Often tools will provide commands to manage dependencies. The following is a
+crude example of modifying the content using the ``tomlkit`` library.
+
+.. code:: python
+
+    import re, tomlkit
+
+    def add(script: str, dependency: str) -> str:
+        match = re.search(r'(?ms)^__pyproject__ *= *"""$(.+?)^"""$', script)
+        config = tomlkit.parse(match.group(1))
+        config['project']['dependencies'].append(dependency)
+
+        start, end = match.span(1)
+        return script[:start] + tomlkit.dumps(config) + script[end:]
+
+
+Backwards Compatibility
+=======================
+
+At the time of writing, ``__pyproject__`` only appears 5 times
+`on GitHub <https://github.com/search?q=__pyproject__&type=code>`__ and 4 of
+those belong to a user who appears to already be using this PEP's format
+for its intended purpose.
+
+
+How to Teach This
+=================
+
+The format is intended to bridge the gap between users just writing scripts and
+users writing packages. Knowledge of how to write metadata for one use case will
+be directly transferable to the other.
+
+Everything in the parent PEP regarding teaching is applicable to this as well.
+
+
+Recommendations
+===============
+
+For situations in which users do not define the required name and version
+fields, the following defaults should be preferred:
+
+* ``name``: ``script-<sha256 of script's path>`` e.g. ``script-3a5c6b...`` to
+  provide interoperability with other tools that use the name to derive file
+  system storage paths for things like virtual environments
+* ``version``: ``0.0.0``
+
+
+Benefits
+========
+
+Ecosystem cohesion
+------------------
+
+One of the central themes we discovered from the recent
+`packaging survey <https://discuss.python.org/t/22420>`__ is that users have
+begun getting frustrated with the lack of unification regarding both tooling
+and specs.
+
+Adding yet another way to define metadata would further fragment the community.
+
+
+Extensibility
+-------------
+
+The parent PEP allows for extensions to the custom metadata block format. This
```

Comment:
Please stop referring to PEP 722 as the "parent PEP".

Reply:
(Here's how to suggest edits: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request)

Reply:
Ta. That "suggest edit" icon wasn't present in the editor. I was editing an existing comment, if that makes a difference (it shouldn't IMO, but that's a github UI matter)?

In general, I find the "suggest edit" functionality is great, but has a nasty habit of going missing way too easily...

Reply:
Yeah, I think it only shows for new comments, not for editing. And you need to be on the "Files changed" tab (and not on a diff for a single commit).

Reply:
```suggestion
:pep:`722` allows for extensions to the custom metadata block format. This
```